Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -203,23 +309,41 @@ private class TopSitesLayout: UICollectionViewLayout {
             let attr = layoutAttributesForItemAtIndexPath(indexPath)
             attrs.append(attr)
 
-            if i >= thumbnailCount - 1 {
-                let decoration = layoutAttributesForDecorationViewOfKind(SeparatorKind, atIndexPath: indexPath)
+            var kind: String
+            if editing && indexPath.item < thumbnailCount {
+                kind = CloseButtonKind
+            } else {
+                kind = SeparatorKind
+            }
+
+            if let decoration = layoutAttributesForDecorationViewOfKind(kind, atIndexPath: indexPath) {
                 attrs.append(decoration)
             }
         }
         return attrs
     }
 
-    // Set the frames for the row separators.
+    // Set the frames for the row separators and close buttons if we are in editing mode.
     override func layoutAttributesForDecorationViewOfKind(elementKind: String, atIndexPath indexPath: NSIndexPath) -> UICollectionViewLayoutAttributes! {
-        let rowIndex = indexPath.item - thumbnailCount + 1
-        let rowYOffset = CGFloat(rowIndex) * AppConstants.DefaultRowHeight
-        let y = topSectionHeight + rowYOffset
-
-        let decoration = UICollectionViewLayoutAttributes(forDecorationViewOfKind: elementKind, withIndexPath: indexPath)
-        decoration.frame = CGRectMake(0, y, width, 0.5)
-        return decoration
+        if editing && indexPath.item < thumbnailCount {
+            let itemFrame = self.layoutAttributesForItemAtIndexPath(indexPath).frame
+            var deleteFrame = CGRect()
+            deleteFrame.size = CGSize(width: 20, height: 20)
+            deleteFrame.center = CGPoint(x: itemFrame.origin.x + TopSitesPanelUX.SuggestedTileImagePadding, y: itemFrame.origin.y + TopSitesPanelUX.SuggestedTileImagePadding)
+            var attrs = UICollectionViewLayoutAttributes(forDecorationViewOfKind: elementKind, withIndexPath: indexPath)
+            attrs.frame = deleteFrame
+            return attrs
+        } else if (indexPath.item - thumbnailCount + 1) >= 0 {
```

Comment:
No need for `else`s since you're returning. And without using `else` here, you'll be able to calculate `rowIndex` before this `if` condition so you don't have to calculate it twice.